hack/lib/util.sh

```diff
@@ -139,6 +135,21 @@ kube::util::gen-doc() {
   ${cmd} "${tmpdir}"
   # create the list of generated files
   ls "${tmpdir}" | LC_ALL=C sort > "${tmpdir}/.files_generated"
+
+  # remove all old generated file from the destination
+  while read file; do
+    if [[ -e "${tmpdir}/${file}" && -n "${skipprefix}" ]]; then
+      local original=$(cat "${dest}/${file}" | grep -v "${skipprefix}")
```

Comment:
Minor problem here with `local` discarding the error result of `$(...)`, so ideally what you want is:

```shell
  local original generated
  original=$(cat "${dest}/${file}" | grep -v "${skipprefix)")
```

Though you actually want that to work even if `grep` fails (which happens if the initial file is empty or made of only lines matching ${skipprefix}!) so you want the `|| :` there. You can also combine the `cat` and `grep` since the `cat` isn't really doing anything there... You could also assume that the prefix is a prefix (though it's still a regexp) and actually add the `^` here.

You also are doing both `original` and `generated` from ${dest}, I think one of them was supposed to be ${tmpdir}.

```shell
  local original generated
  original=$(grep -v "^${skipprefix}" "${tmpdir}/${file}") || :
  generated=$(grep -v "^${skipprefix}" "${dest}/${file}") || :
```